File: 198. House Robber new.py
class Solution(object):
    def rob(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        T = [0 for i in xrange(len(nums) + 1)]
        if len(nums) == 0:
            return 0
        elif len(nums) == 1 or len(nums) == 2:
            return max(nums)
            
        T[1] = nums[0]
        for i in xrange(2, len(nums) + 1):
            T[i] = max(T[i - 1], T[i - 2] + nums[i - 1])
        return T[len(nums)]

File: 198. House Robber new.cpp
class Solution {
public:
    int rob(vector<int>& nums) {
        if (nums.size() == 0) return 0;
        vector <int> T (nums.size() + 1);
        T[1] = nums[0];
        for (int i = 2; i < nums.size() + 1; i++)
            T[i] = T[i - 1] > T[i - 2] + nums[i - 1] ? T[i - 1] : T[i - 2] + nums[i - 1];
        return T[nums.size()];
    }
};
